inception-recommendation/src/test/java/de/tudarmstadt/ukp/inception/recommendation/exporter/RecommenderExporterTest.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2018
+ * Ubiquitous Knowledge Processing (UKP) Lab
+ * Technische Universität Darmstadt
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package de.tudarmstadt.ukp.inception.recommendation.exporter;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+import java.util.zip.ZipFile;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+
+import de.tudarmstadt.ukp.clarin.webanno.api.AnnotationSchemaService;
+import de.tudarmstadt.ukp.clarin.webanno.api.export.ProjectExportRequest;
+import de.tudarmstadt.ukp.clarin.webanno.api.export.ProjectImportRequest;
+import de.tudarmstadt.ukp.clarin.webanno.export.model.ExportedProject;
+import de.tudarmstadt.ukp.clarin.webanno.model.AnnotationLayer;
+import de.tudarmstadt.ukp.clarin.webanno.model.Project;
+import de.tudarmstadt.ukp.inception.recommendation.api.RecommendationService;
+import de.tudarmstadt.ukp.inception.recommendation.api.model.Recommender;
+
+public class RecommenderExporterTest
+{
+
+    private @Mock AnnotationSchemaService annotationService;
+    private @Mock RecommendationService recommendationService;
+    private Project project;
+    private AnnotationLayer layer;
+
+    private RecommenderExporter sut;
+
+    @Before
+    public void setUp()
+    {
+        initMocks(this);
+
+        layer = new AnnotationLayer();
+        layer.setName("Layer");
+
+        project = new Project();
+        project.setName("Test Project");
+        when(recommendationService.listRecommenders(project)).thenReturn(recommenders());
+
+        when(annotationService.getLayer(layer.getName(), project)).thenReturn(layer);
+
+        sut = new RecommenderExporter(annotationService, recommendationService);
+    }
+
+    @Test
+    public void thatExportingWorks()
+    {
+        // Export the project
+        ProjectExportRequest exportRequest = new ProjectExportRequest();
+        exportRequest.setProject(project);
+        ExportedProject exportedProject = new ExportedProject();
+        File file = mock(File.class);
+
+        sut.exportData(exportRequest, exportedProject, file);
```

Comment:
What's a sut? Why not call it "exporter" or something?

Reply:
System under test, to indicate which object we want to test. If you do not like it, then I can rename it for you. https://en.wikipedia.org/wiki/System_under_test

Reply:
Though so ;)

> "The term SUT means also a stage of maturity of the software because a system test is the successor of integration test in the testing cycle."

Not sure if this really fits here. But I kind of understand why the name is used here - seems you want to highlight which of the objects being juggled around is the one which is being tested. I also understand that a unit test should not test more than one thing at a time, so the "sut" should be unique. I don't have a strong opinion.

Reply:
> System under test (SUT) refers to a system that is being tested for correct operation.